.github/workflows/fail-fast.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+if [[ "$(jq --raw-output '.pull_request' "${GITHUB_EVENT_PATH}")" == "null" ]]; then
```

Comment:
care to add just a few comments ? :D Reading this I understand the idea, but I have absolutely no idea what is actually being done, in particular in the tests :D